docs/data/upload.md

```diff
@@ -0,0 +1,116 @@
+
+- Run a Python script to upload files from a folder using the data client API `file_upload_from_path` method.
+- Run a Golang script calling the data management service `sync` method.
+- Upload images from your mobile device using the Viam mobile app.
+
+## Upload data with Python
+
+You can use the Viam Python SDK's data client API [`file_upload_from_path`](/build/program/apis/data-client/#fileuploadfrompath) method to upload one or more files from your computer to the Viam cloud.
+
+1. Install the [Viam Python SDK](https://python.viam.dev/) by running the following command on the computer from which you want to upload data:
+
+   ```sh {class="command-line" data-prompt="$"}
+   pip install viam-sdk
+   ```
+
+2. Create a Python script file in a directory of your choice and [add code to establish a connection](/build/program/apis/data-client/#establish-a-connection) from your computer to your [Viam app](https://app.viam.com) {{< glossary_tooltip term_id="location" text="location" >}} or individual {{< glossary_tooltip term_id="part" text="machine part" >}}.
+
+3. Use the `file_upload_from_path` method to upload your data, depending on whether you are uploading one or multiple files:
+
+   - To upload just one file, make a call to `file_upload_from_path` according to [the data client API documentation](/build/program/apis/data-client/#fileuploadfrompath).
+     The following example code could be placed inside the `main()` function (or a function called from `main()`):
+
+     ```python {class="line-numbers linkable-line-numbers"}
+     await data_client.file_upload_from_path(
+       # The ID of the machine part the file should be associated with
+       part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+       # Any tags you want to apply to this file
+       tags=["cat", "animals", "brown"],
+       # Path to the file
+       filepath="/Users/Artoo/my_cat_photos/brown-cat-on-a-couch.png"
+     )
+     ```
+
+   - To upload all the files in a directory, you can use the same [`file_upload_from_path`](/build/program/apis/data-client/#fileuploadfrompath) method inside a `for` loop, for example:
+
+     ```python {class="line-numbers linkable-line-numbers"}
+     import os # Add this package at the top of your program
+                      # with your other imports
+
+     my_data_directory = "/Users/Artoo/my_cat_photos"
+
+     for file_name in os.listdir(my_data_directory):
+       await data_client.file_upload_from_path(
+         part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+         tags=["cat", "animals", "brown"],
+         filepath=os.path.join(my_data_directory, file_name)
+       )
+     ```
+
+4. Save and run your code once.
+   Running your code more than once will duplicate the data.
+   View your uploaded data in your [**DATA** page in the Viam app](https://app.viam.com/data/view).
+
+## Sync data with Golang
```

Comment:
instruction was to capture how to use sync **to upload files from a folder**. This [sync](https://docs.viam.com/data/cloud-sync/#sync-files-from-another-directory) with an additional files path.